// @voiceflow/base-types/build/cjs/utils/ai.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Role = exports.KNOWLEDGE_BASE_NOT_FOUND = exports.DATA_SOURCE = exports.PROMPT_MODE = exports.ChatModels = exports.GPT_MODEL = void 0;
var GPT_MODEL;
(function (GPT_MODEL) {
    GPT_MODEL["DaVinci_003"] = "text-davinci-003";
    GPT_MODEL["GPT_3_5_turbo"] = "gpt-3.5-turbo";
    GPT_MODEL["GPT_4"] = "gpt-4";
    GPT_MODEL["CLAUDE_V1"] = "claude-v1";
    GPT_MODEL["CLAUDE_V2"] = "claude-v2";
    GPT_MODEL["CLAUDE_INSTANT_V1"] = "claude-instant-v1";
})(GPT_MODEL = exports.GPT_MODEL || (exports.GPT_MODEL = {}));
exports.ChatModels = [GPT_MODEL.GPT_3_5_turbo, GPT_MODEL.GPT_4];
var PROMPT_MODE;
(function (PROMPT_MODE) {
    PROMPT_MODE["PROMPT"] = "prompt";
    PROMPT_MODE["MEMORY"] = "memory";
    PROMPT_MODE["MEMORY_PROMPT"] = "memory_prompt";
})(PROMPT_MODE = exports.PROMPT_MODE || (exports.PROMPT_MODE = {}));
var DATA_SOURCE;
(function (DATA_SOURCE) {
    DATA_SOURCE["KNOWLEDGE_BASE"] = "knowledge_base";
    DATA_SOURCE["DEFAULT"] = "default";
})(DATA_SOURCE = exports.DATA_SOURCE || (exports.DATA_SOURCE = {}));
exports.KNOWLEDGE_BASE_NOT_FOUND = '[not found]';
var Role;
(function (Role) {
    Role["SYSTEM"] = "system";
    Role["ASSISTANT"] = "assistant";
    Role["USER"] = "user";
})(Role = exports.Role || (exports.Role = {}));
