// @voiceflow/base-types/build/esm/node/visual.js
export var DeviceType;
(function (DeviceType) {
    DeviceType["MOBILE"] = "mobile";
    DeviceType["TABLET"] = "tablet";
    DeviceType["DESKTOP"] = "desktop";
    DeviceType["SMART_WATCH"] = "smart_watch";
    DeviceType["TELEVISION"] = "television";
    DeviceType["IN_CAR_DISPLAY"] = "in_car_display";
    DeviceType["ECHO_SPOT"] = "echo_spot";
    DeviceType["ECHO_SHOW_8"] = "echo_show_8";
    DeviceType["ECHO_SHOW_10"] = "echo_show_10";
    DeviceType["FIRE_HD_8"] = "fire_hd_8";
    DeviceType["FIRE_HD_10"] = "fire_hd_10";
    DeviceType["FIRE_TV_CUBE"] = "fire_tv_cube";
    DeviceType["GOOGLE_NEST_HUB"] = "google_nest_hub";
})(DeviceType || (DeviceType = {}));
export var CanvasVisibility;
(function (CanvasVisibility) {
    CanvasVisibility["FULL"] = "full";
    CanvasVisibility["HIDDEN"] = "hidden";
    CanvasVisibility["CROPPED"] = "cropped";
})(CanvasVisibility || (CanvasVisibility = {}));
export var APLType;
(function (APLType) {
    APLType["JSON"] = "JSON";
    APLType["SPLASH"] = "SPLASH";
})(APLType || (APLType = {}));
export var VisualType;
(function (VisualType) {
    VisualType["APL"] = "apl";
    VisualType["IMAGE"] = "image";
})(VisualType || (VisualType = {}));
export var FrameType;
(function (FrameType) {
    FrameType["AUTO"] = "AUTO";
    FrameType["DEVICE"] = "DEVICE";
    FrameType["CUSTOM_SIZE"] = "CUSTOM_SIZE";
})(FrameType || (FrameType = {}));
