// @voiceflow/base-types/build/cjs/models/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Version = exports.VariableState = exports.Transcript = exports.ProjectSecret = exports.Project = exports.Program = exports.Diagram = exports.CustomBlock = exports.ApiKey = void 0;
exports.ApiKey = __importStar(require("./apiKey"));
__exportStar(require("./base"), exports);
exports.CustomBlock = __importStar(require("./customBlock"));
exports.Diagram = __importStar(require("./diagram"));
exports.Program = __importStar(require("./program"));
exports.Project = __importStar(require("./project"));
exports.ProjectSecret = __importStar(require("./projectSecret"));
exports.Transcript = __importStar(require("./transcripts"));
exports.VariableState = __importStar(require("./variableState"));
exports.Version = __importStar(require("./version"));
