// @voiceflow/sdk-runtime/build/cjs/runtime/runtime.interface.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isRuntimeTrace = exports.isRuntimeTraceType = exports.createIntentAction = exports.ActionType = exports.isRuntimeAction = exports.isPublicRuntimeOptions = exports.isAuthRuntimeOptions = exports.Trace = void 0;
const base_types_1 = require("@voiceflow/base-types");
var base_types_2 = require("@voiceflow/base-types");
Object.defineProperty(exports, "Trace", { enumerable: true, get: function () { return base_types_2.Trace; } });
const isAuthRuntimeOptions = (options) => {
    return !!options?.verify?.authorization;
};
exports.isAuthRuntimeOptions = isAuthRuntimeOptions;
const isPublicRuntimeOptions = (options) => {
    return typeof options?.verify?.projectID === 'string';
};
exports.isPublicRuntimeOptions = isPublicRuntimeOptions;
const isRuntimeAction = (value) => typeof value === 'object' && Object.prototype.hasOwnProperty.call(value, 'type');
exports.isRuntimeAction = isRuntimeAction;
exports.ActionType = base_types_1.BaseRequest.RequestType;
const createIntentAction = (payload) => ({
    type: exports.ActionType.INTENT,
    payload,
});
exports.createIntentAction = createIntentAction;
const isRuntimeTraceType = (value) => typeof value === 'string' && Object.values(base_types_1.Trace.TraceType).includes(value);
exports.isRuntimeTraceType = isRuntimeTraceType;
const isRuntimeTrace = (value) => typeof value === 'object' &&
    Object.prototype.hasOwnProperty.call(value, 'type') &&
    (0, exports.isRuntimeTraceType)(value.type);
exports.isRuntimeTrace = isRuntimeTrace;
