// @voiceflow/sdk-runtime/build/esm/runtime/runtime.service.auth.js
import { RuntimeService } from '../runtime/runtime.service.js';
export class AuthRuntimeService extends RuntimeService {
    constructor(options) {
        super(options);
        this.authorization = options.verify.authorization;
    }
    async interact(request) {
        const { action, config, sessionID, versionID } = request;
        return this.send(`state/user/${sessionID}/interact`, {
            method: 'POST',
            body: { action, config },
            headers: {
                authorization: this.authorization,
                sessionID,
                ...(versionID ? { versionID } : {}),
            },
            params: new URLSearchParams({ verbose: 'true' }),
        });
    }
    async feedback() {
        return Promise.reject(new Error('not implemented'));
    }
    // TODO: expose authenticated publishing
    getPublishing() {
        return Promise.reject(new Error('not implemented'));
    }
    // TODO: expose authenticated createTranscript
    createTranscript() {
        return Promise.reject(new Error('not implemented'));
    }
}
