// @voiceflow/base-types/build/esm/version/settings.js
export var RepeatType;
(function (RepeatType) {
    RepeatType["OFF"] = "OFF";
    RepeatType["ALL"] = "ALL";
    RepeatType["DIALOG"] = "DIALOG";
})(RepeatType || (RepeatType = {}));
export var SessionType;
(function (SessionType) {
    SessionType["RESUME"] = "resume";
    SessionType["RESTART"] = "restart";
})(SessionType || (SessionType = {}));
export var GlobalNoMatchType;
(function (GlobalNoMatchType) {
    GlobalNoMatchType["STATIC"] = "static";
    GlobalNoMatchType["GENERATIVE"] = "generative";
})(GlobalNoMatchType || (GlobalNoMatchType = {}));
export const defaultSettings = ({ error = null, repeat = RepeatType.ALL, session = { type: SessionType.RESTART }, defaultCanvasNodeVisibility = null, defaultCarouselLayout = null, globalNoMatch = { type: GlobalNoMatchType.STATIC, prompt: undefined }, globalNoReply = { delay: undefined, prompt: undefined }, } = {}) => ({
    error,
    repeat,
    session,
    defaultCanvasNodeVisibility,
    defaultCarouselLayout,
    globalNoMatch,
    globalNoReply,
});
