// @voiceflow/base-types/build/esm/runtimeLogs/logs/kinds.js
import { NodeType } from "../../node";
/** Similar to {@link NodeType}, but for runtime logging */
export var StepLogKind;
(function (StepLogKind) {
    // Response
    StepLogKind["TEXT"] = "text";
    StepLogKind["SPEAK"] = "speak";
    StepLogKind["AUDIO"] = "audio";
    StepLogKind["VISUALS"] = "visuals";
    StepLogKind["CARD"] = "card";
    StepLogKind["CAROUSEL"] = "carousel";
    // User input
    StepLogKind["BUTTONS"] = "buttons";
    StepLogKind["CHOICE"] = "choice";
    StepLogKind["CAPTURE"] = "capture";
    StepLogKind["PROMPT"] = "prompt";
    StepLogKind["INTENT"] = "intent";
    // Logic
    StepLogKind["CONDITION"] = "condition";
    StepLogKind["SET"] = "set";
    StepLogKind["RANDOM"] = "random";
    StepLogKind["FLOW"] = "flow";
    StepLogKind["EXIT"] = "exit";
    // Integration
    StepLogKind["API"] = "api";
    /** @deprecated Will be removed soon */
    StepLogKind["GOOGLE_SHEETS"] = "google_sheets";
    StepLogKind["CUSTOM_CODE"] = "custom_code";
    StepLogKind["CUSTOM_ACTION"] = "custom_action";
    // Special
    StepLogKind["START"] = "start";
})(StepLogKind || (StepLogKind = {}));
export var GlobalLogKind;
(function (GlobalLogKind) {
    GlobalLogKind["CONVERSATION_START"] = "conversation_start";
    GlobalLogKind["NLU_INTENT_RESOLVED"] = "nlu.intent_resolved";
})(GlobalLogKind || (GlobalLogKind = {}));
const NODE_TYPE_TO_STEP_LOG_KIND = {
    [NodeType.TEXT]: StepLogKind.TEXT,
    [NodeType.SPEAK]: StepLogKind.SPEAK,
    [NodeType.START]: StepLogKind.START,
    [NodeType.CARD]: StepLogKind.CARD,
    [NodeType.CAROUSEL]: StepLogKind.CAROUSEL,
    [NodeType.BUTTONS]: StepLogKind.BUTTONS,
    [NodeType.SET]: StepLogKind.SET,
    [NodeType.SET_V2]: StepLogKind.SET,
    [NodeType.IF]: StepLogKind.CONDITION,
    [NodeType.IF_V2]: StepLogKind.CONDITION,
    [NodeType.RANDOM]: StepLogKind.RANDOM,
    [NodeType.CAPTURE]: StepLogKind.CAPTURE,
    [NodeType.CAPTURE_V2]: StepLogKind.CAPTURE,
    [NodeType.API]: StepLogKind.API,
    /** @deprecated */
    [NodeType.GOOGLE_SHEETS]: StepLogKind.GOOGLE_SHEETS,
    [NodeType.INTENT]: StepLogKind.INTENT,
    [NodeType.CODE]: StepLogKind.CUSTOM_CODE,
    [NodeType.EXIT]: StepLogKind.EXIT,
    [NodeType.PROMPT]: StepLogKind.PROMPT,
    [NodeType.VISUAL]: StepLogKind.VISUALS,
};
// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
export function nodeTypeToStepLogKind(nodeType) {
    return nodeType in NODE_TYPE_TO_STEP_LOG_KIND ? NODE_TYPE_TO_STEP_LOG_KIND[nodeType] : undefined;
}
