// @voiceflow/common/build/cjs/utils/string.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.conditionalReplace = exports.removeTrailingUnderscores = exports.stripHTMLTags = exports.arrayStringReplace = exports.capitalizeAllWords = exports.capitalizeFirstLetter = void 0;
const TAGS_REGEXP = /(<([^>]+)>)/gi;
const TRAILING_UNDERSCORES_REGEXP = /(^_+)|(_+$)/g;
const capitalizeFirstLetter = (value) => value.charAt(0).toUpperCase() + value.slice(1);
exports.capitalizeFirstLetter = capitalizeFirstLetter;
const capitalizeAllWords = (value) => value.split(' ').map(exports.capitalizeFirstLetter).join(' ');
exports.capitalizeAllWords = capitalizeAllWords;
const arrayStringReplace = (targetString, newString, stringArray) => stringArray.map((string) => string.replace(targetString, newString));
exports.arrayStringReplace = arrayStringReplace;
const stripHTMLTags = (str) => str.replace(TAGS_REGEXP, '');
exports.stripHTMLTags = stripHTMLTags;
const removeTrailingUnderscores = (str) => str.replace(TRAILING_UNDERSCORES_REGEXP, '');
exports.removeTrailingUnderscores = removeTrailingUnderscores;
const conditionalReplace = (base, pattern, value) => {
    return value ? base.replace(pattern, value) : base;
};
exports.conditionalReplace = conditionalReplace;
