// @voiceflow/base-types/build/cjs/models/project/knowledgeBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChunkStrategyType = exports.KnowledgeBaseDocumentStatus = exports.KnowledgeBaseDocumentType = void 0;
var KnowledgeBaseDocumentType;
(function (KnowledgeBaseDocumentType) {
    KnowledgeBaseDocumentType["PDF"] = "pdf";
    KnowledgeBaseDocumentType["TEXT"] = "text";
    KnowledgeBaseDocumentType["URL"] = "url";
    KnowledgeBaseDocumentType["DOCX"] = "docx";
})(KnowledgeBaseDocumentType = exports.KnowledgeBaseDocumentType || (exports.KnowledgeBaseDocumentType = {}));
var KnowledgeBaseDocumentStatus;
(function (KnowledgeBaseDocumentStatus) {
    KnowledgeBaseDocumentStatus["ERROR"] = "ERROR";
    KnowledgeBaseDocumentStatus["SUCCESS"] = "SUCCESS";
    KnowledgeBaseDocumentStatus["PENDING"] = "PENDING";
    KnowledgeBaseDocumentStatus["INITIALIZED"] = "INITIALIZED";
})(KnowledgeBaseDocumentStatus = exports.KnowledgeBaseDocumentStatus || (exports.KnowledgeBaseDocumentStatus = {}));
var ChunkStrategyType;
(function (ChunkStrategyType) {
    ChunkStrategyType["RECURSIVE_TEXT_SPLITTER"] = "recursive_text_splitter";
})(ChunkStrategyType = exports.ChunkStrategyType || (exports.ChunkStrategyType = {}));
