// @voiceflow/common/build/esm/utils/normalized.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { stringify } from './functional';
import { hasProperty } from './object';
const createEmptyNormalized = () => ({ byKey: {}, allKeys: [] });
export const EMPTY = createEmptyNormalized();
export const defaultGetKey = (obj) => stringify(obj.id);
/**
 * @deprecated prefer `normal-store`
 */
export const safeAdd = (items, obj) => (items.includes(obj) ? items : [...items, obj]);
const safeAddToStart = (items, obj) => (items.includes(obj) ? items : [obj, ...items]);
export const getByIndex = (items) => (_, index) => items[index];
export const getByKey = (lookup) => (key) => lookup[key];
export const buildLookup = (allKeys, getValue) => allKeys.reduce((acc, key, index) => {
    acc[key] = getValue(key, index);
    return acc;
}, {});
/**
 * @deprecated prefer `normal-store`
 */
export const normalize = (items, getKey) => {
    const allKeys = items.map(getKey !== null && getKey !== void 0 ? getKey : defaultGetKey);
    return {
        byKey: buildLookup(allKeys, getByIndex(items)),
        allKeys,
    };
};
/**
 * @deprecated prefer `normal-store`
 */
export const denormalize = ({ allKeys, byKey }) => allKeys.map((key) => byKey[key]);
/**
 * @deprecated prefer `normal-store`
 */
export const getNormalizedByKey = ({ byKey }, key) => byKey[key];
/**
 * @deprecated prefer `normal-store`
 */
export const safeGetNormalizedByKey = ({ byKey }, key) => hasProperty(byKey, key) ? byKey[key] : null;
/**
 * @deprecated prefer `normal-store`
 */
export const getAllNormalizedByKeys = ({ byKey }, keys) => keys.map((key) => byKey[key]);
/**
 * @deprecated prefer `normal-store`
 */
export const updateNormalizedByKey = (_a, key, obj) => {
    var { byKey } = _a, rest = __rest(_a, ["byKey"]);
    return (Object.assign(Object.assign({}, rest), { byKey: Object.assign(Object.assign({}, byKey), { [key]: obj }) }));
};
/**
 * @deprecated prefer `normal-store`
 */
export const addNormalizedByKey = (normalized, key, obj) => (Object.assign(Object.assign({}, updateNormalizedByKey(normalized, key, obj)), { allKeys: safeAdd(normalized.allKeys, key) }));
/**
 * @deprecated prefer `normal-store`
 */
export const addToStartNormalizedByKey = (normalized, key, obj) => (Object.assign(Object.assign({}, updateNormalizedByKey(normalized, key, obj)), { allKeys: safeAddToStart(normalized.allKeys, key) }));
/**
 * @deprecated prefer `normal-store`
 */
export const addAllNormalizedByKeys = (normalized, objs, getKey) => {
    const keyGetter = getKey !== null && getKey !== void 0 ? getKey : defaultGetKey;
    const keys = objs.map(keyGetter);
    return {
        byKey: Object.assign(Object.assign({}, normalized.byKey), objs.reduce((acc, obj) => Object.assign(acc, { [keyGetter(obj)]: obj }), {})),
        allKeys: keys.reduce(safeAdd, normalized.allKeys),
    };
};
/**
 * @deprecated prefer `normal-store`
 */
export const reorderKeys = (newKeyArray, byKey) => ({
    allKeys: newKeyArray,
    byKey,
});
