// @voiceflow/eslint-config/config/base.js
module.exports = {
  extends: [
    'airbnb-base', // include import rules
    'plugin:sonarjs/recommended',
    'plugin:promise/recommended',
    'plugin:eslint-comments/recommended',
    'plugin:you-dont-need-lodash-underscore/compatible',
  ],
  plugins: ['sonarjs', 'promise', 'prefer-arrow', 'no-secrets', 'you-dont-need-lodash-underscore', 'unicorn'],
  parserOptions: {
    ecmaVersion: 9,
  },
  // https://eslint.org/docs/user-guide/configuring/rules#report-unused-eslint-disable-comments
  reportUnusedDisableDirectives: true,
  rules: {
    // error
    'arrow-parens': ['error', 'always'],
    'callback-return': 'error',
    'capitalized-comments': 'error',
    'handle-callback-err': 'error',
    'max-depth': ['error', 3],
    'max-len': [
      'error',
      {
        code: 150,
        ignoreUrls: true,
        ignoreStrings: true,
        ignoreTemplateLiterals: true,
        ignoreRegExpLiterals: true,
      },
    ],
    'max-nested-callbacks': ['error', 5],
    'max-params': ['error', 5],
    'no-console': 'error',
    'no-process-exit': 'error',
    'no-useless-call': 'error',
    'quote-props': ['error', 'as-needed'],
    'prefer-arrow-callback': 'error',
    'lines-between-class-members': 'error',

    'prefer-arrow/prefer-arrow-functions': [
      'error',
      {
        disallowPrototype: false,
        singleReturnOnly: true,
        classPropertiesAllowed: false,
      },
    ],

    'eslint-comments/no-unused-disable': 'error',
    'eslint-comments/disable-enable-pair': ['error', { allowWholeFile: true }],

    'no-secrets/no-secrets': 'error',

    'unicorn/better-regex': 'error',
    'unicorn/prefer-set-has': 'error',

    // warn
    'sonarjs/no-duplicate-string': 'warn',

    // off
    'no-plusplus': 'off',
    'no-continue': 'off',
    'require-jsdoc': 'off', // not sure we want this
    quotes: ['error', 'single', 'avoid-escape'],
    'valid-jsdoc': 'off',
    'func-names': 'off',
    'class-methods-use-this': 'off',
    'import/prefer-default-export': 'off',
    camelcase: 'off',
    'no-underscore-dangle': 'off',
  },
};
