// @voiceflow/eslint-config/typescript.js
module.exports = {
  parser: '@typescript-eslint/parser',
  plugins: ['@typescript-eslint'],
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      extends: ['plugin:@typescript-eslint/recommended'],
      rules: {
        // error
        '@typescript-eslint/no-empty-interface': [
          'error',
          {
            // Allow `interface T extends U {}`
            allowSingleExtends: true,
          },
        ],
        '@typescript-eslint/consistent-type-definitions': ['error', 'interface'],

        // off
        'no-shadow': 'off',
        'no-unused-vars': 'off',
        'no-use-before-define': 'off',
        'no-useless-constructor': 'off',

        '@typescript-eslint/explicit-module-boundary-types': 'off',
        '@typescript-eslint/no-non-null-assertion': 'off',
        '@typescript-eslint/no-explicit-any': 'off',

        'import/no-unresolved': 'off',
        'import/export': 'off',
        'import/named': 'off',
      },
    },
  ],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      typescript: {
        alwaysTryTypes: true,

        project: 'tsconfig.json',
      },
    },
  },
};
