// @voiceflow/base-types/build/cjs/utils/node.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.isIntegrations = exports.isInteraction = exports.isCaptureV2 = exports.isDirective = exports.isGoToNode = exports.isRandomV2 = exports.isCarousel = exports.isGeneral = exports.isCapture = exports.isCardV2 = exports.isRandom = exports.isStream = exports.isVisual = exports.isSetV2 = exports.isStart = exports.isSpeak = exports.isExit = exports.isCode = exports.isFlow = exports.isIfV2 = exports.isCard = exports.isGoTo = exports.isText = exports.isSet = exports.isUrl = exports.isIf = exports.isV1 = exports.createNodeTypeguard = void 0;
const Node = __importStar(require("../node"));
const common_1 = require("@voiceflow/common");
exports.createNodeTypeguard = common_1.Utils.typeguard.createTypedTypeguardCreator();
const isV1 = (node) => common_1.Utils.object.hasProperty(node, '_v') && node._v === 1;
exports.isV1 = isV1;
exports.isIf = (0, exports.createNodeTypeguard)(Node.NodeType.IF);
exports.isUrl = (0, exports.createNodeTypeguard)(Node.NodeType.URL);
exports.isSet = (0, exports.createNodeTypeguard)(Node.NodeType.SET);
exports.isText = (0, exports.createNodeTypeguard)(Node.NodeType.TEXT);
exports.isGoTo = (0, exports.createNodeTypeguard)(Node.NodeType.GOTO);
exports.isCard = (0, exports.createNodeTypeguard)(Node.NodeType.CARD);
exports.isIfV2 = (0, exports.createNodeTypeguard)(Node.NodeType.IF_V2);
exports.isFlow = (0, exports.createNodeTypeguard)(Node.NodeType.FLOW);
exports.isCode = (0, exports.createNodeTypeguard)(Node.NodeType.CODE);
exports.isExit = (0, exports.createNodeTypeguard)(Node.NodeType.EXIT);
exports.isSpeak = (0, exports.createNodeTypeguard)(Node.NodeType.SPEAK);
exports.isStart = (0, exports.createNodeTypeguard)(Node.NodeType.START);
exports.isSetV2 = (0, exports.createNodeTypeguard)(Node.NodeType.SET_V2);
exports.isVisual = (0, exports.createNodeTypeguard)(Node.NodeType.VISUAL);
exports.isStream = (0, exports.createNodeTypeguard)(Node.NodeType.STREAM);
exports.isRandom = (0, exports.createNodeTypeguard)(Node.NodeType.RANDOM);
exports.isCardV2 = (0, exports.createNodeTypeguard)(Node.NodeType.CARD_V2);
exports.isCapture = (0, exports.createNodeTypeguard)(Node.NodeType.CAPTURE);
exports.isGeneral = (0, exports.createNodeTypeguard)(Node.NodeType.GENERAL);
exports.isCarousel = (0, exports.createNodeTypeguard)(Node.NodeType.CAROUSEL);
exports.isRandomV2 = (0, exports.createNodeTypeguard)(Node.NodeType.RANDOM_V2);
exports.isGoToNode = (0, exports.createNodeTypeguard)(Node.NodeType.GOTO_NODE);
exports.isDirective = (0, exports.createNodeTypeguard)(Node.NodeType.DIRECTIVE);
exports.isCaptureV2 = (0, exports.createNodeTypeguard)(Node.NodeType.CAPTURE_V2);
exports.isInteraction = (0, exports.createNodeTypeguard)(Node.NodeType.INTERACTION);
exports.isIntegrations = (0, exports.createNodeTypeguard)(Node.NodeType.INTEGRATIONS);
