// @voiceflow/base-types/build/esm/utils/ai.js
export var GPT_MODEL;
(function (GPT_MODEL) {
    GPT_MODEL["DaVinci_003"] = "text-davinci-003";
    GPT_MODEL["GPT_3_5_turbo"] = "gpt-3.5-turbo";
    GPT_MODEL["GPT_4"] = "gpt-4";
    GPT_MODEL["CLAUDE_V1"] = "claude-v1";
    GPT_MODEL["CLAUDE_V2"] = "claude-v2";
    GPT_MODEL["CLAUDE_INSTANT_V1"] = "claude-instant-v1";
})(GPT_MODEL || (GPT_MODEL = {}));
export const ChatModels = [GPT_MODEL.GPT_3_5_turbo, GPT_MODEL.GPT_4];
export var PROMPT_MODE;
(function (PROMPT_MODE) {
    PROMPT_MODE["PROMPT"] = "prompt";
    PROMPT_MODE["MEMORY"] = "memory";
    PROMPT_MODE["MEMORY_PROMPT"] = "memory_prompt";
})(PROMPT_MODE || (PROMPT_MODE = {}));
export var DATA_SOURCE;
(function (DATA_SOURCE) {
    DATA_SOURCE["KNOWLEDGE_BASE"] = "knowledge_base";
    DATA_SOURCE["DEFAULT"] = "default";
})(DATA_SOURCE || (DATA_SOURCE = {}));
export const KNOWLEDGE_BASE_NOT_FOUND = '[not found]';
export var Role;
(function (Role) {
    Role["SYSTEM"] = "system";
    Role["ASSISTANT"] = "assistant";
    Role["USER"] = "user";
})(Role || (Role = {}));
