// @voiceflow/voiceflow-types/build/cjs/constants/platformType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TTSPlatformType = exports.ProjectType = exports.PlatformType = void 0;
var PlatformType;
(function (PlatformType) {
    PlatformType["ALEXA"] = "alexa";
    PlatformType["GOOGLE"] = "google";
    PlatformType["VOICEFLOW"] = "voiceflow";
    PlatformType["DIALOGFLOW_ES"] = "df-es";
    PlatformType["DIALOGFLOW_CX"] = "df-cx";
    PlatformType["LUIS"] = "luis";
    PlatformType["RASA"] = "rasa";
    PlatformType["WATSON"] = "watson";
    PlatformType["LEX"] = "lex";
    PlatformType["EINSTEIN"] = "einstein";
    PlatformType["NUANCE_MIX"] = "nuance-mix";
    PlatformType["WEBCHAT"] = "webchat";
    PlatformType["MICROSOFT_TEAMS"] = "microsoft_teams";
    PlatformType["WHATSAPP"] = "whatsapp";
    PlatformType["SMS"] = "sms";
    /** @deprecated use VOICEFLOW instead */
    PlatformType["IVR"] = "ivr";
    /** @deprecated use VOICEFLOW instead */
    PlatformType["CHATBOT"] = "chatbot";
    /** @deprecated use DIALOGFLOW_ES instead */
    PlatformType["DIALOGFLOW_ES_CHAT"] = "dialogflow_es_chat";
    /** @deprecated use VOICEFLOW instead */
    PlatformType["DIALOGFLOW_ES_VOICE"] = "dialogflow_es_voice";
    /** @deprecated use VOICEFLOW instead */
    PlatformType["GENERAL"] = "general";
    /** @deprecated will be removed soon */
    PlatformType["MOBILE_APP"] = "mobile_app";
})(PlatformType = exports.PlatformType || (exports.PlatformType = {}));
var ProjectType;
(function (ProjectType) {
    ProjectType["CHAT"] = "chat";
    ProjectType["VOICE"] = "voice";
})(ProjectType = exports.ProjectType || (exports.ProjectType = {}));
var TTSPlatformType;
(function (TTSPlatformType) {
    TTSPlatformType["ALEXA"] = "alexa";
    TTSPlatformType["AZURE"] = "azure";
    TTSPlatformType["GOOGLE"] = "google";
})(TTSPlatformType = exports.TTSPlatformType || (exports.TTSPlatformType = {}));
