// @voiceflow/common/build/cjs/utils/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.variables = exports.typeguard = exports.timezones = exports.time = exports.string = exports.slot = exports.protocol = exports.promise = exports.object = exports.number = exports.normalized = exports.mathjs = exports.map = exports.intent = exports.id = exports.generateHash = exports.generate = exports.functional = exports.emails = exports.array = void 0;
exports.array = __importStar(require("./array"));
exports.emails = __importStar(require("./emails"));
exports.functional = __importStar(require("./functional"));
var generate_1 = require("./generate");
Object.defineProperty(exports, "generate", { enumerable: true, get: function () { return generate_1.generate; } });
Object.defineProperty(exports, "generateHash", { enumerable: true, get: function () { return generate_1.generateHash; } });
exports.id = __importStar(require("./id"));
exports.intent = __importStar(require("./intent"));
exports.map = __importStar(require("./map"));
exports.mathjs = __importStar(require("./mathjs"));
exports.normalized = __importStar(require("./normalized"));
exports.number = __importStar(require("./number"));
exports.object = __importStar(require("./object"));
exports.promise = __importStar(require("./promise"));
exports.protocol = __importStar(require("./protocol"));
exports.slot = __importStar(require("./slot"));
exports.string = __importStar(require("./string"));
exports.time = __importStar(require("./time"));
exports.timezones = __importStar(require("./timezones"));
exports.typeguard = __importStar(require("./typeguard"));
exports.variables = __importStar(require("./variables"));
// TODO: Remove these exports once we migrate all utils in other repos.
__exportStar(require("./intent"), exports);
__exportStar(require("./variables"), exports);
