// @voiceflow/base-types/build/cjs/utils/step.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.isGoogleSheets = exports.isInteraction = exports.isDeprecated = exports.isGoToDomain = exports.isCaptureV2 = exports.isComponent = exports.isDirective = exports.isGoToNode = exports.isGeneral = exports.isCapture = exports.isButtons = exports.isCommand = exports.isRandomV2 = exports.isRandom = exports.isZapier = exports.isStream = exports.isIntent = exports.isVisual = exports.isCarousel = exports.isPrompt = exports.isSetV2 = exports.isStart = exports.isSpeak = exports.isExit = exports.isCode = exports.isGoTo = exports.isFlow = exports.isIfV2 = exports.isCardV2 = exports.isCard = exports.isText = exports.isApi = exports.isSet = exports.isUrl = exports.isIf = exports.createStepTypeguard = void 0;
const Node = __importStar(require("../node"));
const common_1 = require("@voiceflow/common");
exports.createStepTypeguard = common_1.Utils.typeguard.createTypedTypeguardCreator();
exports.isIf = (0, exports.createStepTypeguard)(Node.NodeType.IF);
exports.isUrl = (0, exports.createStepTypeguard)(Node.NodeType.URL);
exports.isSet = (0, exports.createStepTypeguard)(Node.NodeType.SET);
exports.isApi = (0, exports.createStepTypeguard)(Node.NodeType.API);
exports.isText = (0, exports.createStepTypeguard)(Node.NodeType.TEXT);
exports.isCard = (0, exports.createStepTypeguard)(Node.NodeType.CARD);
exports.isCardV2 = (0, exports.createStepTypeguard)(Node.NodeType.CARD_V2);
exports.isIfV2 = (0, exports.createStepTypeguard)(Node.NodeType.IF_V2);
exports.isFlow = (0, exports.createStepTypeguard)(Node.NodeType.FLOW);
exports.isGoTo = (0, exports.createStepTypeguard)(Node.NodeType.GOTO);
exports.isCode = (0, exports.createStepTypeguard)(Node.NodeType.CODE);
exports.isExit = (0, exports.createStepTypeguard)(Node.NodeType.EXIT);
exports.isSpeak = (0, exports.createStepTypeguard)(Node.NodeType.SPEAK);
exports.isStart = (0, exports.createStepTypeguard)(Node.NodeType.START);
exports.isSetV2 = (0, exports.createStepTypeguard)(Node.NodeType.SET_V2);
exports.isPrompt = (0, exports.createStepTypeguard)(Node.NodeType.PROMPT);
exports.isCarousel = (0, exports.createStepTypeguard)(Node.NodeType.CAROUSEL);
exports.isVisual = (0, exports.createStepTypeguard)(Node.NodeType.VISUAL);
exports.isIntent = (0, exports.createStepTypeguard)(Node.NodeType.INTENT);
exports.isStream = (0, exports.createStepTypeguard)(Node.NodeType.STREAM);
exports.isZapier = (0, exports.createStepTypeguard)(Node.NodeType.ZAPIER);
exports.isRandom = (0, exports.createStepTypeguard)(Node.NodeType.RANDOM);
exports.isRandomV2 = (0, exports.createStepTypeguard)(Node.NodeType.RANDOM_V2);
exports.isCommand = (0, exports.createStepTypeguard)(Node.NodeType.COMMAND);
exports.isButtons = (0, exports.createStepTypeguard)(Node.NodeType.BUTTONS);
exports.isCapture = (0, exports.createStepTypeguard)(Node.NodeType.CAPTURE);
exports.isGeneral = (0, exports.createStepTypeguard)(Node.NodeType.GENERAL);
exports.isGoToNode = (0, exports.createStepTypeguard)(Node.NodeType.GOTO_NODE);
exports.isDirective = (0, exports.createStepTypeguard)(Node.NodeType.DIRECTIVE);
exports.isComponent = (0, exports.createStepTypeguard)(Node.NodeType.COMPONENT);
exports.isCaptureV2 = (0, exports.createStepTypeguard)(Node.NodeType.CAPTURE_V2);
exports.isGoToDomain = (0, exports.createStepTypeguard)(Node.NodeType.GOTO_DOMAIN);
exports.isDeprecated = (0, exports.createStepTypeguard)(Node.NodeType.DEPRECATED);
exports.isInteraction = (0, exports.createStepTypeguard)(Node.NodeType.INTERACTION);
exports.isGoogleSheets = (0, exports.createStepTypeguard)(Node.NodeType.GOOGLE_SHEETS);
