// @voiceflow/base-types/build/cjs/runtimeLogs/logs/kinds.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nodeTypeToStepLogKind = exports.GlobalLogKind = exports.StepLogKind = void 0;
const node_1 = require("../../node");
/** Similar to {@link NodeType}, but for runtime logging */
var StepLogKind;
(function (StepLogKind) {
    // Response
    StepLogKind["TEXT"] = "text";
    StepLogKind["SPEAK"] = "speak";
    StepLogKind["AUDIO"] = "audio";
    StepLogKind["VISUALS"] = "visuals";
    StepLogKind["CARD"] = "card";
    StepLogKind["CAROUSEL"] = "carousel";
    // User input
    StepLogKind["BUTTONS"] = "buttons";
    StepLogKind["CHOICE"] = "choice";
    StepLogKind["CAPTURE"] = "capture";
    StepLogKind["PROMPT"] = "prompt";
    StepLogKind["INTENT"] = "intent";
    // Logic
    StepLogKind["CONDITION"] = "condition";
    StepLogKind["SET"] = "set";
    StepLogKind["RANDOM"] = "random";
    StepLogKind["FLOW"] = "flow";
    StepLogKind["EXIT"] = "exit";
    // Integration
    StepLogKind["API"] = "api";
    /** @deprecated Will be removed soon */
    StepLogKind["GOOGLE_SHEETS"] = "google_sheets";
    StepLogKind["CUSTOM_CODE"] = "custom_code";
    StepLogKind["CUSTOM_ACTION"] = "custom_action";
    // Special
    StepLogKind["START"] = "start";
})(StepLogKind = exports.StepLogKind || (exports.StepLogKind = {}));
var GlobalLogKind;
(function (GlobalLogKind) {
    GlobalLogKind["CONVERSATION_START"] = "conversation_start";
    GlobalLogKind["NLU_INTENT_RESOLVED"] = "nlu.intent_resolved";
})(GlobalLogKind = exports.GlobalLogKind || (exports.GlobalLogKind = {}));
const NODE_TYPE_TO_STEP_LOG_KIND = {
    [node_1.NodeType.TEXT]: StepLogKind.TEXT,
    [node_1.NodeType.SPEAK]: StepLogKind.SPEAK,
    [node_1.NodeType.START]: StepLogKind.START,
    [node_1.NodeType.CARD]: StepLogKind.CARD,
    [node_1.NodeType.CAROUSEL]: StepLogKind.CAROUSEL,
    [node_1.NodeType.BUTTONS]: StepLogKind.BUTTONS,
    [node_1.NodeType.SET]: StepLogKind.SET,
    [node_1.NodeType.SET_V2]: StepLogKind.SET,
    [node_1.NodeType.IF]: StepLogKind.CONDITION,
    [node_1.NodeType.IF_V2]: StepLogKind.CONDITION,
    [node_1.NodeType.RANDOM]: StepLogKind.RANDOM,
    [node_1.NodeType.CAPTURE]: StepLogKind.CAPTURE,
    [node_1.NodeType.CAPTURE_V2]: StepLogKind.CAPTURE,
    [node_1.NodeType.API]: StepLogKind.API,
    /** @deprecated */
    [node_1.NodeType.GOOGLE_SHEETS]: StepLogKind.GOOGLE_SHEETS,
    [node_1.NodeType.INTENT]: StepLogKind.INTENT,
    [node_1.NodeType.CODE]: StepLogKind.CUSTOM_CODE,
    [node_1.NodeType.EXIT]: StepLogKind.EXIT,
    [node_1.NodeType.PROMPT]: StepLogKind.PROMPT,
    [node_1.NodeType.VISUAL]: StepLogKind.VISUALS,
};
// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
function nodeTypeToStepLogKind(nodeType) {
    return nodeType in NODE_TYPE_TO_STEP_LOG_KIND ? NODE_TYPE_TO_STEP_LOG_KIND[nodeType] : undefined;
}
exports.nodeTypeToStepLogKind = nodeTypeToStepLogKind;
