// @voiceflow/base-types/build/esm/utils/step.js
import * as Node from "../node";
import { Utils } from '@voiceflow/common';
export const createStepTypeguard = Utils.typeguard.createTypedTypeguardCreator();
export const isIf = createStepTypeguard(Node.NodeType.IF);
export const isUrl = createStepTypeguard(Node.NodeType.URL);
export const isSet = createStepTypeguard(Node.NodeType.SET);
export const isApi = createStepTypeguard(Node.NodeType.API);
export const isText = createStepTypeguard(Node.NodeType.TEXT);
export const isCard = createStepTypeguard(Node.NodeType.CARD);
export const isCardV2 = createStepTypeguard(Node.NodeType.CARD_V2);
export const isIfV2 = createStepTypeguard(Node.NodeType.IF_V2);
export const isFlow = createStepTypeguard(Node.NodeType.FLOW);
export const isGoTo = createStepTypeguard(Node.NodeType.GOTO);
export const isCode = createStepTypeguard(Node.NodeType.CODE);
export const isExit = createStepTypeguard(Node.NodeType.EXIT);
export const isSpeak = createStepTypeguard(Node.NodeType.SPEAK);
export const isStart = createStepTypeguard(Node.NodeType.START);
export const isSetV2 = createStepTypeguard(Node.NodeType.SET_V2);
export const isPrompt = createStepTypeguard(Node.NodeType.PROMPT);
export const isCarousel = createStepTypeguard(Node.NodeType.CAROUSEL);
export const isVisual = createStepTypeguard(Node.NodeType.VISUAL);
export const isIntent = createStepTypeguard(Node.NodeType.INTENT);
export const isStream = createStepTypeguard(Node.NodeType.STREAM);
export const isZapier = createStepTypeguard(Node.NodeType.ZAPIER);
export const isRandom = createStepTypeguard(Node.NodeType.RANDOM);
export const isRandomV2 = createStepTypeguard(Node.NodeType.RANDOM_V2);
export const isCommand = createStepTypeguard(Node.NodeType.COMMAND);
export const isButtons = createStepTypeguard(Node.NodeType.BUTTONS);
export const isCapture = createStepTypeguard(Node.NodeType.CAPTURE);
export const isGeneral = createStepTypeguard(Node.NodeType.GENERAL);
export const isGoToNode = createStepTypeguard(Node.NodeType.GOTO_NODE);
export const isDirective = createStepTypeguard(Node.NodeType.DIRECTIVE);
export const isComponent = createStepTypeguard(Node.NodeType.COMPONENT);
export const isCaptureV2 = createStepTypeguard(Node.NodeType.CAPTURE_V2);
export const isGoToDomain = createStepTypeguard(Node.NodeType.GOTO_DOMAIN);
export const isDeprecated = createStepTypeguard(Node.NodeType.DEPRECATED);
export const isInteraction = createStepTypeguard(Node.NodeType.INTERACTION);
export const isGoogleSheets = createStepTypeguard(Node.NodeType.GOOGLE_SHEETS);
