// @voiceflow/base-types/build/cjs/node/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RUNTIME_ONLY_NODES = exports.NodeType = void 0;
var NodeType;
(function (NodeType) {
    NodeType["TEXT"] = "text";
    NodeType["SPEAK"] = "speak";
    NodeType["START"] = "start";
    NodeType["CARD"] = "card";
    NodeType["CARD_V2"] = "cardV2";
    NodeType["CAROUSEL"] = "carousel";
    NodeType["BUTTONS"] = "buttons";
    NodeType["INTERACTION"] = "interaction";
    // ai
    NodeType["AI_RESPONSE"] = "generative";
    NodeType["AI_SET"] = "ai_set";
    // logic
    NodeType["SET"] = "set";
    NodeType["SET_V2"] = "setV2";
    NodeType["IF"] = "if";
    NodeType["IF_V2"] = "ifV2";
    NodeType["RANDOM"] = "random";
    NodeType["RANDOM_V2"] = "randomV2";
    NodeType["CAPTURE"] = "capture";
    NodeType["CAPTURE_V2"] = "captureV2";
    // navigation
    NodeType["GOTO"] = "goTo";
    NodeType["GOTO_NODE"] = "goToNode";
    NodeType["GOTO_DOMAIN"] = "goToDomain";
    // integrations
    NodeType["API"] = "api";
    NodeType["ZAPIER"] = "zapier";
    NodeType["INTEGRATIONS"] = "integrations";
    NodeType["GOOGLE_SHEETS"] = "google_sheets";
    // advanced
    NodeType["INTENT"] = "intent";
    NodeType["FLOW"] = "flow";
    NodeType["COMPONENT"] = "component";
    NodeType["CODE"] = "code";
    NodeType["EXIT"] = "exit";
    NodeType["PROMPT"] = "prompt";
    NodeType["COMMAND"] = "command";
    NodeType["TRACE"] = "trace";
    NodeType["CHANNEL_ACTION"] = "channel_action";
    NodeType["CUSTOM_BLOCK_POINTER"] = "custom_block_pointer";
    // other
    NodeType["URL"] = "url";
    NodeType["STREAM"] = "stream";
    NodeType["VISUAL"] = "visual";
    NodeType["GENERAL"] = "general";
    NodeType["DIRECTIVE"] = "directive";
    NodeType["DEPRECATED"] = "deprecated";
})(NodeType = exports.NodeType || (exports.NodeType = {}));
exports.RUNTIME_ONLY_NODES = [NodeType.GOTO];
