// @voiceflow/voiceflow-types/build/esm/luis/types.js
import { SlotType } from '../constants';
export var BuiltInLuisSlotType;
(function (BuiltInLuisSlotType) {
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-age */
    BuiltInLuisSlotType["AGE"] = "age";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-currency */
    BuiltInLuisSlotType["CURRENCY"] = "money";
    /** @deprecated @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-deprecated */
    BuiltInLuisSlotType["DATETIME"] = "datetime";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-datetimev2 */
    BuiltInLuisSlotType["DATETIME_V2"] = "datetimeV2";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-dimension */
    BuiltInLuisSlotType["DIMENSION"] = "dimension";
    /** @deprecated @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-deprecated#encyclopedia-culture */
    BuiltInLuisSlotType["ENCYCLOPEDIA"] = "encyclopedia";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-email */
    BuiltInLuisSlotType["EMAIL"] = "email";
    /** @deprecated @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-deprecated#geography-culture */
    BuiltInLuisSlotType["GEOGRAPHY"] = "geography";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-geographyv2 */
    BuiltInLuisSlotType["GEOGRAPHY_V2"] = "geographyV2";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-keyphrase */
    BuiltInLuisSlotType["KEY_PHRASE"] = "keyPhrase";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-number */
    BuiltInLuisSlotType["NUMBER"] = "number";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-ordinal */
    BuiltInLuisSlotType["ORDINAL"] = "ordinal";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-ordinal-v2 */
    BuiltInLuisSlotType["ORDINAL_V2"] = "ordinalV2";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-percentage */
    BuiltInLuisSlotType["PERCENTAGE"] = "percentage";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-person */
    BuiltInLuisSlotType["PERSON_NAME"] = "personName";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-phonenumber */
    BuiltInLuisSlotType["PHONENUMBER"] = "phonenumber";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-temperature */
    BuiltInLuisSlotType["TEMPERATURE"] = "temperature";
    /** @see https://docs.microsoft.com/en-us/azure/cognitive-services/luis/luis-reference-prebuilt-url */
    BuiltInLuisSlotType["URL"] = "url";
})(BuiltInLuisSlotType || (BuiltInLuisSlotType = {}));
// Maps general type to Luis compatible format
export const GENERAL_SLOT_TYPE_TO_LUIS = new Map([
    [SlotType.AGE, BuiltInLuisSlotType.AGE],
    [SlotType.CURRENCY, BuiltInLuisSlotType.CURRENCY],
    [SlotType.DATETIME, BuiltInLuisSlotType.DATETIME_V2],
    [SlotType.DIMENSION, BuiltInLuisSlotType.DIMENSION],
    [SlotType.EMAIL, BuiltInLuisSlotType.EMAIL],
    [SlotType.GEOGRAPHY, BuiltInLuisSlotType.GEOGRAPHY_V2],
    [SlotType.KEY_PHRASE, BuiltInLuisSlotType.KEY_PHRASE],
    [SlotType.NAME, BuiltInLuisSlotType.PERSON_NAME],
    [SlotType.NUMBER, BuiltInLuisSlotType.NUMBER],
    [SlotType.ORDINAL, BuiltInLuisSlotType.ORDINAL],
    // ORDINAL_V2 doesn't exist in VoiceflowConstants.SlotType currently, might get modified in the future once ORDINAL_V2 gets better language support on LUIS
    // eslint-disable-next-line no-secrets/no-secrets
    // [VoiceflowConstants.SlotType.ORDINAL_V2, BuiltInLuisSlotType.ORDINAL_V2],
    [SlotType.PERCENTAGE, BuiltInLuisSlotType.PERCENTAGE],
    [SlotType.PHONENUMBER, BuiltInLuisSlotType.PHONENUMBER],
    [SlotType.TEMPERATURE, BuiltInLuisSlotType.TEMPERATURE],
    [SlotType.URL, BuiltInLuisSlotType.URL],
]);
