// @voiceflow/base-types/build/cjs/text/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isLinkElement = exports.isVariableElement = exports.ElementProperty = exports.ElementType = exports.TextProperty = void 0;
var TextProperty;
(function (TextProperty) {
    TextProperty["BACKGROUND_COLOR"] = "backgroundColor";
    TextProperty["COLOR"] = "color";
    TextProperty["ITALIC"] = "italic";
    TextProperty["UNDERLINE"] = "underline";
    TextProperty["FONT_WEIGHT"] = "fontWeight";
    TextProperty["FONT_FAMILY"] = "fontFamily";
    TextProperty["STRIKE_THROUGH"] = "strikeThrough";
})(TextProperty = exports.TextProperty || (exports.TextProperty = {}));
var ElementType;
(function (ElementType) {
    ElementType["LINK"] = "link";
    ElementType["VARIABLE"] = "variable";
})(ElementType = exports.ElementType || (exports.ElementType = {}));
var ElementProperty;
(function (ElementProperty) {
    ElementProperty["TEXT_ALIGN"] = "textAlign";
})(ElementProperty = exports.ElementProperty || (exports.ElementProperty = {}));
const isVariableElement = (element) => {
    return element.type === ElementType.VARIABLE;
};
exports.isVariableElement = isVariableElement;
const isLinkElement = (element) => {
    return element.type === ElementType.LINK;
};
exports.isLinkElement = isLinkElement;
